### PR TITLE
site status management for movies

### PR DESCRIPTION
**Pull request by mikolajgraczyk**
I'm temporarily moving dispatches to the pagination component (because currently it only renders once) so that they only run once. When they are in the PopularMoviesPage and PopularPeoplePage components, they are called unnecessarily every time these components are rendered. (When, for example, we change tabs from people to movies).

### CODE REVIEW

File: src/features/movies/moviesSaga.js [hunk 1]
```diff
@@ -14,16 +14,15 @@ function* fetchGenresHandler() {
...
   }
 }
 
 function* fetchMoviesHandler() {
   try {
     const movies = yield call(getPopularData, "movie");
-    yield put(setMovies(movies));
     yield delay(1000);
-    yield put(setFetchingToSucces());
+    yield put(setFetchingToSucces(movies));
```
Comment (marcin10lw) [inline src/features/movies/moviesSaga.js:24]: Nice you made it into a single put 💪

File: src/features/movies/moviesSlice.js [hunk 2]
```diff
@@ -41,9 +39,15 @@ export const selectFetchingStatus = (state) =>
   selectMoviesState(state).fetchingStatus;
 
 export const selectedGenreByIds = (state, genreIds) => {
+  const genres = selectGenres(state);
+
+  if (genres.length > 1) {
```
Comment (marcin10lw) [inline src/features/movies/moviesSlice.js]: I think it should be `genres.length > 0`

File: src/features/movies/moviesSlice.js [hunk 3]
```diff
@@ -41,9 +39,15 @@ export const selectFetchingStatus = (state) =>
...
     return genreIds.map((genreId) => {
```
Comment (marcin10lw) [inline src/features/movies/moviesSlice.js]: During making of actor page i've realized that some of cast movies return genre_ids array with ids that don't match any id in our genres
Now when we map over genreIds it always returns array with the same length and we can get an array like this:
`[undefined, 123, undefined]`.
Later when can't map over such array on Tags.
I suggest changing our selectGenresByIds to look like this:
image

**Review by marcin10lw — CHANGES_REQUESTED** [review]
You did a great job! We just need to update our selectGenresByIds

**Review by marcin10lw — APPROVED** [review]

**Review by HelkaPL — APPROVED** [review]
It's looks like legit.
Let's merge it that way.

**Pull request merged**